### PR TITLE
New IA - Overview pages

### PR DESCRIPTION
**Pull request by DavidRollins**
### For internal users - Please add a Jira DX PR ticket to the subject!
<br>

https://deploy-preview-5700--tyk-docs.netlify.app/docs/nightly/tyk-overview/
https://deploy-preview-5700--tyk-docs.netlify.app/docs/nightly/tyk-components/
https://deploy-preview-5700--tyk-docs.netlify.app/docs/nightly/plan-your-api-integration/

---

### Preview Link

<br>

### Description
1. Describe your changes in detail. Why is this change required? What problem does it solve?
     2. Please explain the change to the reviewer. Pay special attention to changes that are hard to spot
       2.1. Update overview section with overview pages:
           2.1.1 Tyk Overview - introduction to Tyk and the problems Tyk looks to help businesses solve
           2.1.2 Tyk Components - What makes up Tyk and intro to how they work
           2.1.3 Planning phase details - centralizing planning your API
     3. @mentions of the person to review the proposed changes. They need to be able to know the topic well in order to approve it.
<br>

#### Screenshots (if appropriate)
<br>

## Checklist

- [ ] I have added a **preview link** to the PR description.
- [ ] I have reviewed the suggestions made by our AI (PR Agent) and updated them accordingly (spelling errors, rephrasing, etc.)
- [ ] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [ ] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [ ] Make sure you have started *your change* off *our latest `master`*.
- [ ] I **labeled** the PR

___

### **PR Type**
Documentation

___

### **Description**
- Added new documentation pages for Tyk Overview, Tyk Components, and Plan Your API Integration.
- The new pages provide detailed information on Tyk's API management capabilities and components.
- Updated the menu configuration to include links to the new overview pages.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>plan-your-api-integration.md</strong><dd><code>Add detailed guide for planning API integration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/plan-your-api-integration.md

<li>Added a comprehensive guide on planning API integration with Tyk.<br> <li> Included sections on API purpose, design, documentation, development, <br>deployment, and monitoring.<br> <li> Highlighted Tyk's support for various API lifecycle stages.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5700/files#diff-f117bfad324561c7a2444f1826d0f92bef6997abaae276d9dacfcc5ad5d39080">+113/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>tyk-components.md</strong><dd><code>Introduce and explain Tyk's API management components</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/tyk-components.md

<li>Introduced Tyk's API management components.<br> <li> Detailed the roles of each component in API management.<br> <li> Provided links to further documentation for each component.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5700/files#diff-bbb23ad40b5eb51a6e05c1d3da65d52dc0aff00edab70a7a9da9a0ce5c51280b">+216/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>tyk-overview.md</strong><dd><code>Provide an overview of Tyk's API management</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/tyk-overview.md

<li>Provided an overview of Tyk's API management capabilities.<br> <li> Explained the need for API management and Tyk's solutions.<br> <li> Described how Tyk's components work together.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5700/files#diff-33af4f663a9d2d0b94d78df1ae2092f72ee9608bbbd14b5bb3126301ea5b1f5b">+67/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>menu.yaml</strong><dd><code>Update menu to include new overview pages</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/data/menu.yaml

<li>Updated menu structure to include new overview pages.<br> <li> Added links to Tyk Overview, Tyk Components, and Plan Your API <br>Integration.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5700/files#diff-a6789c66bfa0660f7fdbe4796002ab4777677bae59a0276689c840100ff0b0b2">+17/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

**Review by sharadregoti — APPROVED** [review]

File: tyk-docs/content/tyk-components.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+{{< badge title="Tyk Gateway" href="tyk-components#tyk-gateway" >}} The open-source core of Tyk that handles request routing, rate-limiting, and caching. {{< /badge >}}
+
+{{< badge title="Tyk Dashboard" href="tyk-components#tyk-dashboard" >}} A powerful interface for managing and monitoring APIs, users, and permissions. {{< /badge >}}
+
+{{< badge title="Tyk Enterprise Developer Portal" href="tyk-components#tyk-enterprise-developer-portal" >}} A customizable portal for enabling API discovery and user self-service. {{< /badge >}}
+
+{{< badge title="Tyk Multi Data Centre Bridge" href="tyk-components#tyk-multi-data-centre-bridge-mdcb" >}} Ensures high availability by synchronizing APIs across multiple data centers. {{< /badge >}}
+
+{{< badge title="Tyk Pump" href="tyk-components#tyk-pump" >}} Pushes analytics data from the Gateway to external data stores for advanced insights. {{< /badge >}}
```
Comment (carolwkc) [inline tyk-docs/content/tyk-components.md]:
```suggestion
{{< badge title="Tyk Pump" href="tyk-components#tyk-pump" >}} Aggregates and exports API analytics from Gateway to enable insights and advanced monitoring. {{< /badge >}}
```

File: tyk-docs/content/tyk-components.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+{{< badge title="Tyk Operator" href="tyk-components#tyk-operator" >}} Integrates with Kubernetes for seamless API management within containerized environments. {{< /badge >}}
+
+{{< badge title="Tyk Streams" href="tyk-components#tyk-streams" >}} Enables real-time event streaming for APIs, ideal for push notifications and live data. {{< /badge >}}
+
+{{< badge title="Tyk Sync" href="tyk-components#tyk-sync" >}} Automates the synchronization of API configurations across environments. {{< /badge >}}
```
Comment (carolwkc) [inline tyk-docs/content/tyk-components.md]:
```suggestion
{{< badge title="Tyk Sync" href="tyk-components#tyk-sync" >}} Synchronizes API configurations across environments for consistency and streamlined deployments. {{< /badge >}}
```

File: tyk-docs/content/tyk-components.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+
+{{< grid >}}
+
+{{< badge title="Tyk Gateway" href="tyk-components#tyk-gateway" >}} The open-source core of Tyk that handles request routing, rate-limiting, and caching. {{< /badge >}}
+
+{{< badge title="Tyk Dashboard" href="tyk-components#tyk-dashboard" >}} A powerful interface for managing and monitoring APIs, users, and permissions. {{< /badge >}}
+
+{{< badge title="Tyk Enterprise Developer Portal" href="tyk-components#tyk-enterprise-developer-portal" >}} A customizable portal for enabling API discovery and user self-service. {{< /badge >}}
+
+{{< badge title="Tyk Multi Data Centre Bridge" href="tyk-components#tyk-multi-data-centre-bridge-mdcb" >}} Ensures high availability by synchronizing APIs across multiple data centers. {{< /badge >}}
```
Comment (carolwkc) [inline tyk-docs/content/tyk-components.md]:
```suggestion
{{< badge title="Tyk Multi Data Centre Bridge" href="tyk-components#tyk-multi-data-centre-bridge-mdcb" >}} Enables centralized API management across distributed data centers for global reach and compliance. {{< /badge >}}
```

File: tyk-docs/content/tyk-components.md [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+MDCB is designed for organizations with extensive geographic reach, supporting both global distribution and local compliance.
+
+---
+
+### Tyk Pump
+
+**[Tyk Pump]({{< ref "tyk-pump" >}})** is a lightweight tool that aggregates API analytics from the Tyk Gateway and exports them to various storage systems for analysis:
+
+- **Data Collection**: Continuously gathers real-time API analytics.
+- **Data Export**: Integrates with multiple backends like MongoDB, AWS S3, and others.
```
Comment (carolwkc) [inline tyk-docs/content/tyk-components.md]:
```suggestion
- **Data Export**: Integrates with multiple backends like Splunk, Prometheus, ElasticSearch, and others.
```

File: tyk-docs/content/tyk-components.md [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+The UDG simplifies complex data interactions, enabling developers to work with a unified GraphQL endpoint for various backend services.
+
+---
+
+## How Tyk’s Components Work Together
+
+Tyk’s components create a unified API management ecosystem, covering all aspects of API lifecycle management:
+
+- **Traffic and Security**: The Tyk Gateway acts as the first line of defense, handling traffic and enforcing security protocols.
+- **Configuration and Deployment**: Tyk Dashboard, Tyk Operator, and Tyk Helm Charts provide tools for managing, deploying, and scaling APIs.
```
Comment (carolwkc) [inline tyk-docs/content/tyk-components.md]:
```suggestion
- **Configuration and Deployment**: Tyk Dashboard, Tyk Operator, and Tyk Sync provide tools for managing, deploying, and scaling APIs.
```

File: tyk-docs/content/tyk-components.md [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+- **Developer Experience**: The Tyk Developer Portal enhances the developer experience, promoting API discovery and user engagement.
+- **Global Reach and Resilience**: MDCB and Tyk Sync ensure consistent API configurations and high availability across regions.
```
Comment (carolwkc) [inline tyk-docs/content/tyk-components.md]:
```suggestion
- **Global Reach and Resilience**: MDCB ensures consistent API configurations and high availability across regions.
```

File: tyk-docs/content/tyk-components.md [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+
+---
+
+## How Tyk’s Components Work Together
+
+Tyk’s components create a unified API management ecosystem, covering all aspects of API lifecycle management:
+
+- **Traffic and Security**: The Tyk Gateway acts as the first line of defense, handling traffic and enforcing security protocols.
+- **Configuration and Deployment**: Tyk Dashboard, Tyk Operator, and Tyk Helm Charts provide tools for managing, deploying, and scaling APIs.
+- **Developer Experience**: The Tyk Developer Portal enhances the developer experience, promoting API discovery and user engagement.
```
Comment (carolwkc) [inline tyk-docs/content/tyk-components.md]:
```suggestion
- **Developer Experience**: The Tyk Developer Portal and Identity Broker enhances the developer experience, promoting API discovery and user engagement.
```

File: tyk-docs/content/tyk-components.md [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+- **Global Reach and Resilience**: MDCB and Tyk Sync ensure consistent API configurations and high availability across regions.
+- **Data and Insights**: Tyk Pump and Tyk Streams support real-time monitoring and data streaming, enabling data-driven decisions.
+- **Unified Access**: UDG and Identity Broker provide unified data and identity management, simplifying complex API interactions.
```
Comment (carolwkc) [inline tyk-docs/content/tyk-components.md]:
```suggestion
- **Unified Access**: UDG and Tyk Streams unify API Management by enabling seamless, multi-protocol data integration and delivery.
- **Kubernetes Deployment**: Tyk Helm Charts simplifies deployment of Tyk on Kubernetes.
```

File: tyk-docs/content/tyk-components.md [hunk 9]
```diff
@@ -0,0 +1,216 @@
...
+
+## How Tyk’s Components Work Together
+
+Tyk’s components create a unified API management ecosystem, covering all aspects of API lifecycle management:
+
+- **Traffic and Security**: The Tyk Gateway acts as the first line of defense, handling traffic and enforcing security protocols.
+- **Configuration and Deployment**: Tyk Dashboard, Tyk Operator, and Tyk Helm Charts provide tools for managing, deploying, and scaling APIs.
+- **Developer Experience**: The Tyk Developer Portal enhances the developer experience, promoting API discovery and user engagement.
+- **Global Reach and Resilience**: MDCB and Tyk Sync ensure consistent API configurations and high availability across regions.
+- **Data and Insights**: Tyk Pump and Tyk Streams support real-time monitoring and data streaming, enabling data-driven decisions.
```
Comment (carolwkc) [inline tyk-docs/content/tyk-components.md]:
```suggestion
- **Data and Insights**: Tyk Gateway's native support for OpenTelemetry, combined with Tyk Pump, enables real-time monitoring and analytics.
```

File: tyk-docs/content/plan-your-api-integration.md [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+
+Tyk simplifies the early stages of API planning by providing:
+- **Multi-API Support**: Tyk supports REST, GraphQL, and gRPC APIs, so you can choose the best style for your project.
+- **Advanced Security Options**: Easily configure [authentication methods]({{< ref "basic-config-and-security" >}}) and [permissions]({{< ref "basic-config-and-security/security/dashboard/user-roles" >}}) to protect data and control access.
+- **Developer Portals**: [Tyk’s developer portals]({{< ref "tyk-developer-portal/tyk-portal-classic" >}}) allow you to publish documentation and manage developer access, making it easier for teams or external partners to use your APIs.
+
+#### **Deployment and Configuration Phase**
+
+In this stage, Tyk streamlines deployment, whether on the cloud, on-premises, or hybrid:
+- **Kubernetes Integration with Tyk Operator**: If your API runs on microservices, [Tyk Operator]({{< ref "tyk-operator" >}}) integrates with Kubernetes to help you manage deployments as Kubernetes resources.
```
Comment (carolwkc) [inline tyk-docs/content/plan-your-api-integration.md]:
```suggestion
- **Kubernetes Integration with Tyk Operator**: If your API runs on Kubernetes, [Tyk Operator]({{< ref "tyk-operator" >}}) integrates with Kubernetes to help you manage deployments as Kubernetes resources.
```

File: tyk-docs/content/plan-your-api-integration.md [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+- **Custom Domain Setup**: Configure custom domains to make your API URLs user-friendly and secure with SSL/TLS certificates.
+- **API Gateway Deployment**: [Tyk’s gateways]({{< ref "tyk-oss-gateway" >}}) let you deploy, monitor, and secure multiple APIs from a single platform, so you can [manage environments]({{< ref "advanced-configuration/manage-multiple-environments" >}}) (development, staging, production) with ease.
```
Comment (carolwkc) [inline tyk-docs/content/plan-your-api-integration.md]: Do you mean Tyk Cloud?

Comment (sharadregoti) [reply tyk-docs/content/plan-your-api-integration.md]: Let's rephrase it to `Multiple Environment Management`

File: tyk-docs/content/plan-your-api-integration.md [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+- **Developer Portals**: [Tyk’s developer portals]({{< ref "tyk-developer-portal/tyk-portal-classic" >}}) allow you to publish documentation and manage developer access, making it easier for teams or external partners to use your APIs.
+
+#### **Deployment and Configuration Phase**
+
+In this stage, Tyk streamlines deployment, whether on the cloud, on-premises, or hybrid:
+- **Kubernetes Integration with Tyk Operator**: If your API runs on Kubernetes, [Tyk Operator]({{< ref "api-management/automations#what-is-tyk-operator" >}}) integrates with Kubernetes to help you manage deployments as Kubernetes resources.
+- **Custom Domain Setup**: Configure custom domains to make your API URLs user-friendly and secure with SSL/TLS certificates.
+- **API Gateway Deployment**: [Tyk’s gateways]({{< ref "tyk-oss-gateway" >}}) let you deploy, monitor, and secure multiple APIs from a single platform, so you can [manage environments]({{< ref "advanced-configuration/manage-multiple-environments" >}}) (development, staging, production) with ease.
```
Comment (sharadregoti) [inline tyk-docs/content/plan-your-api-integration.md]:
```suggestion
- **Deployment Across Multiple Environments**: [Tyk’s gateways]({{< ref "tyk-oss-gateway" >}}) let you deploy, monitor, and secure multiple APIs from a single platform, so you can [manage environments]({{< ref "advanced-configuration/manage-multiple-environments" >}}) (development, staging, production) with ease.
```

**Review by sharadregoti — APPROVED** [review]

**Pull request merged**